includes/discovery/sensors/temperature/comware.inc.php

```diff
@@ -8,11 +8,18 @@
  * Free Software Foundation, either version 3 of the License, or (at your
  * option) any later version.  Please see LICENSE.txt at the top level of
  * the source code distribution for details.
- */
+ *
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+*/
+use App\Models\EntPhysical;
 
 echo 'Comware ';
 
-$entphydata = dbFetchRows("SELECT `entPhysicalIndex`, `entPhysicalClass`, `entPhysicalName` FROM `entPhysical` WHERE `device_id` = ? AND `entPhysicalClass` REGEXP 'module|sensor' ORDER BY `entPhysicalIndex`", [$device['device_id']]);
+$entphydata = EntPhysical::where('device_id', $device['device_id'])
```

Comment:
This is only a partial fix because $device needs to go away too.

`DeviceCache::getPrimary()->entityPhysical()` ... (then where entPhysicalClass etc)

Reply:
A not for those who don't know ->entityPhysical will return a collection of the relationship items and ->entityPhysical() will return a query for the relationship items.